zipline/test_algorithms.py

```diff
@@ -784,6 +784,11 @@ def handle_data(context, data):
     pass
 """
 
+no_handle_data = """
+def initialize(context):
+    pass
```

Comment:
I see what you're saying, but I don't think it's worth the trouble to have those requirements.  If you have neither `handle_data` nor a scheduled function, your strategy just won't do anything - which while not very useful, is ok I think.